test/test_framework.py

```diff
@@ -546,38 +548,45 @@ class CustomObject:
         else:
             self.fail("AttributeError not raised")
 
-        obj.state.dict = {}
-        obj.state.dict["a"] = {}
-        obj.state.dict["a"]["b"] = "c"
-        obj.state.dict["a"]["d"] = "e"
-        del obj.state.dict["a"]["d"]
+        framework.commit()
+        framework.close()
+
+        # Test and validation functions in a tuple of 2-tuples. Assignment and keywords like del are not supported in lambdas so functions are used instead.
+        test_funcs = ((
+            lambda: setattr(obj.state, 'dict', {}),
+            lambda: self.assertEqual(dict(obj.state.dict), {})
+        ), (
+            lambda: obj.state.dict.update({"a": {}}),
+            lambda: self.assertEqual(dict(obj.state.dict), {"a": {}})
+        ), (
+            lambda: obj.state.dict["a"].update({"b": "c"}),
+            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+        ), (
+            lambda: obj.state.dict["a"].update({"d": "e"}),
+            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c", "d": "e"}})
+        ), (
+            lambda: obj.state.dict["a"].pop('d'), lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+        ), )
 
-        self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+        for do, validate in test_funcs:
+            framework = self.create_framework()
 
-        self.assertEqual(obj.changes, 5)
+            obj = SomeObject(framework, "1")
 
-        obj.changes = 0
+            do()
+            validate()
 
-        obj.state.list = []
-        obj.state.list.append("a")
-        obj.state.list.append("c")
-        obj.state.list.insert(1, "b")
-        obj.state.list.insert(2, "d")
-        del obj.state.list[2]
+            framework.commit()
+            framework.close()
```

Comment:
What's the benefit of closing/reopening this one every time?

Reply:
I thought of it as a simulation of everything that would be done in a hook execution. But then if we have "mid-hook" commits it becomes more relevant to test this without calling `close`.